Platform/TigerlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1024,6 +1024,8 @@ BoardInit (
   case ReadyToBoot:
     if ((GetBootMode() != BOOT_ON_FLASH_UPDATE) && (GetPayloadId() == 0)) {
       ProgramSecuritySetting ();
+    } else if (GetBootMode() == BOOT_ON_FLASH_UPDATE) {
```

Comment:
why do we only clear it for FWU path ?

Reply:
if we don't clear the flags in FWU path, after FWU, the flags (e.g., BAD DSO), may still exist (warm boot path). Then boot flow will bypass the DSO tuning in new boot.

Reply:
I meant why don't we clear it on all boot path regardless to make it simpler ?  Do you see any issue ?

Reply:
Current logic allows that user only needs to experience 200-sec WDT for each cold boot (if BAD DSO exists).
In current logic, we keep the BAD DSO flag exists, so that in next boot, stage1b can know the existence of BAD DSO and skip Tcc tuning. Otherwise, user will experience 200-sec WDT for every warm boot.

It is OKAY, in terms of solving this infinite reset loop issue, to clear it on all boot paths. However, it is a bit ambiguous, in terms of programming simplification and UX. Please advise.

Reply:
OK.